src/utils/apiHelpers.ts

```diff
@@ -111,3 +111,30 @@ export function moveCodapDrag(context: string, attrTitle: string, mouseX: number
 export function endCodapDrag(context: string, attrTitle: string, mouseX: number, mouseY: number) {
   continueCodapDrag("dragEnd", context, attrTitle, mouseX, mouseY);
 }
+
+export async function getPluginId(): Promise<number | undefined> {
+  try {
+    const response: any = await codapInterface.sendRequest({
+      action: "get",
+      resource: "componentList"
+    });
+    if (response.success) {
+      const plugin = response.values.find((component: any) =>
+        component.type === "game" && component.title === "multidata-plugin"
+      );
+      return plugin?.id;
+    } else {
+      console.error("Error getting plugin ID:", response);
+    }
+  } catch(error) {
+    console.error("Error in getPluginId:", error);
+  }
+  return undefined;
```

Comment:
You could use `get interactiveFrame` instead of searching through the `componentList` here. I'll also point out that this code won't work if the user changes the title of the plugin.